hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsOutputStream.java

```diff
@@ -493,6 +494,12 @@ public synchronized void close() throws IOException {
     }
 
     try {
+      // Check if Executor Service got shutdown before the writes could be
+      // completed.
+      if (hasActiveBlockDataToUpload() && executorService.isShutdown()) {
+        throw new IOException("Executor Service closed before writes could be"
```

Comment:
throw a PathIOException with the output stream path; gives better diagnostics about what failed to be written. whoever fields support calls will appreciate this. I think L515 needs the same treatment.

now, when this is thrown there's no need to catch and wrap it as it is not from flushInternal, so need a way to avoid this. maybe add some boolean on L495 "exceptionCreated", which, if true, allows you to throw the exception on L509 without wrapping.

Reply:
Okay. I intentionally left it to be IOE with no path, since that was being added inside of `wrapException()` anyways. 
Here's how it looked 
```
exception: Failed with java.io.IOException while processing file/directory :[/test/testAbfsThreadPool5ed597ff7617] in method:[Executor Service closed before writes could be completed.]
```
Also, since in `wrapException()` we explicitly check if this is an instance PathIOE, we simply throw it back, we may not require a boolean.